python/flow_sdk/shim_airbyte_cdk.py

```diff
@@ -97,18 +92,21 @@ def to_flow_state(self, index: dict[t.Tuple[str | None, str], t.List[int]], bind
     re-generate the index here as it's possible that binding indices have changed
     """
     @staticmethod
-    def from_flow_state(state: t.Dict[str, t.Any], bindings: t.List[CaptureBinding]):
-        state = State(**{
-            "bindingStateV1": state.get("bindingStateV1", {})
-        })
-
-        index: dict[t.Tuple[str | None, str], t.List[int]] = {}
-        for i, binding in enumerate(bindings):
-            if binding["stateKey"] in state.bindingStateV1:
-                namespace = binding["resourceConfig"].get("namespace",None)
-                stream = binding["resourceConfig"]["stream"]
-                index[(namespace, stream)] = state.bindingStateV1[binding["stateKey"]].rowId
-        return (state, index)
+    def from_flow_state(state: dict | None, bindings: t.List[CaptureBinding]):
+        try:
+            parsed_state = state and State(**state) or State()
```

Comment:
oh neat, this will recursively hydrate the structure from raw `dict` ?

Reply:
Yep! `AirbyteStateMessage`, `StreamState` and all